src/components/Ratings/Ratings.st.css

```diff
@@ -0,0 +1,184 @@
+    -st-named: color, fallback, font;
+}
+
+:import {
+    -st-from: "../../common/palette.st.css";
+    -st-named: errorColor;
+}
+
+:vars {
+    /*Defaults*/
+    DefaultTextColor: color-5;
+    DefaultColor: color-8;
+    DefaultEmptyColor: color-3;
+    DisabledColor: color-3;
+    IconGap: 1.5px;
+    IconLargeGap: 5.5px;
+    IconLargeSize: 30px;
+    DefaultFont: font("{theme: 'Body-S', size: '14px', lineHeight: '1.4em'}");
+    LargeFont: font("{theme: 'Heading-M', size: '32px', lineHeight: '1.25em'}");
+}
+
+:vars {
+    /* Overrides */
+    TextColor: --overridable;
+    IconColor: --overridable;
+    IconEmptyColor: --overridable;
+}
+
+.root {
+    -st-states: disabled, error, size( enum(small, large) ), mode( enum(input, display) ), layout( enum(aside, bottom) );
+
+    display: inline-flex;
+    align-items: center;
+    cursor: pointer;
+}
+
+.root:layout(bottom) {
+    flex-direction: column;
+    align-items: baseline;
+}
+
+/* ------ States ------ */
+
+/* SIZE */
+.root:size(large) .icon {
+   padding: 0 value(IconLargeGap);
+}
+
+.root:size(large) .icon svg {
+    width: value(IconLargeSize);
+    height: value(IconLargeSize);
+}
+
+
+.root .iconList {
+    display: flex;
+    align-items: center;
+    flex-direction: row-reverse;
+    flex-wrap: wrap;
+
+    margin: 0 calc(0px - value(IconGap));
+
+    direction: ltr;
+}
+
+.root:size(large) .iconList {
+    margin: 0 calc(0px - value(IconLargeGap));
+}
+
+.root:size(large) .inputOption,
+.root:size(large) .ratingInfo {
+    font: value(LargeFont);
+}
+
+.root:size(large) .icon {
+    margin-top: 7px;
+}
+
+/* DISABLED */
+.root:hover:not(:disabled):not(:mode(display)) .icon path {
+    stroke: color(fallback(value(IconColor), value(DefaultColor)));
+}
+
+.root:disabled .icon path {
+    stroke: color(value(DisabledColor));
+    fill: transparent;
+};
+
+.root:disabled,
+.root:disabled {
+    cursor: default;
+    pointer-events: none;
+}
+
+/* ERROR */
+.root:error .icon path {
+    fill: transparent;
+    stroke: color(value(errorColor))
+}
+
+/* MODE */
+.root:mode(display),
+.root:mode(display) .icon {
+    cursor: default;
+    pointer-events: none;
+}
+
+.root:mode(display) .icon path {
+    stroke: color(fallback(value(IconEmptyColor), value(DefaultEmptyColor)));
+    fill: color(fallback(value(IconEmptyColor), value(DefaultEmptyColor)));
+}
+
+/* ------ /States -------*/
+
+.icon {
+    -st-states: checked;
+
+    margin-top: 5px;
+    padding: 0 value(IconGap);
+}
+
+.root .icon path {
+    stroke: color(fallback(value(IconColor), value(DefaultColor)));
+}
+
+
+.root:not(:focus-within) .icon:checked path {
+    stroke: color(fallback(value(IconColor), value(DefaultColor)));
+    fill: color(fallback(value(IconColor), value(DefaultColor)));
+ }
+
+.root:not(:mode(display)):hover .icon:hover path,
+.root:not(:mode(display)):hover .icon:hover ~ .icon path,
+.root:not(:mode(display)) .icon:focus-within path,
+.root:not(:mode(display)) .icon:focus-within ~ .icon path {
```

Comment:
regarding `focus-within` i'm not 100% sure we can use it, because I think it's not supported in ie11 and edge.
If we can't use `focus`, then we might need to use the solution written here: https://stackoverflow.com/a/50788621/891850
(also, there's a similar solution in the `Tab.st.css` file, which is part of the `Tabs` component)